django-server/server/admin/static/admin/stats.js

```diff
@@ -0,0 +1,873 @@
+/* Graph settings */
+const YEARS_TO_COVER = 8;
+const MONTHS_TO_COVER = 12;
+const WEEKS_TO_COVER = 4;
+const DAYS_TO_COVER = 14;
+const HOURS_TO_COVER = 48;
+
+const DatasetGrouping = {
+  YEAR: 'year',
+  MONTH: 'month',
+  WEEK: 'week',
+  DAY: 'day',
+  HOUR: 'hour'
+}
+
+const DatasetType = {
+  OCCUPANCY: 'occupancy',
+  RELATIVE_OCCUPANCY: 'relative_occupancy',
+  AVERAGE_DISH_REVIEW: 'average_dish_review',
+}
+
+/* Translation messages */
+const messages = {
+  en: {
+    message: {
+      add_dataset: 'Add dataset',
+      add_first_dataset: 'Add first dataset',
+      add: 'Add',
+      cafeteria: 'Cafeteria',
+      datasets: 'Datasets',
+      delete: 'Delete',
+      group_by: 'Group by',
+      incorect_dataset_type: 'Incorrect dataset type',
+      lt_previous: '< Previous',
+      next_gt: 'Next >',
+      type: 'Type'
+    },
+    dataset_grouping: {
+      'year': 'Year',
+      'month': 'Month',
+      'week': 'Week',
+      'day': 'Day',
+      'hour': 'Hour'
+    },
+    dataset_type: {
+      'occupancy': 'Occupancy',
+      'relative_occupancy': 'Relative occupancy',
+      'average_dish_review': 'Average dish review'
+    }
+  },
+  pl: {
+    message: {
+      add: 'Dodaj',
+      add_dataset: 'Dodaj dane',
+      add_first_dataset: 'Dodaj pierwszy zbiór danych',
+      cafeteria: 'Stołówka',
+      datasets: 'Dane',
+      delete: 'Usuń',
+      group_by: 'Grupuj według',
+      incorect_dataset_type: 'Nieprawidłowy typ danych',
+      lt_previous: '< Poprzednie',
+      next_gt: 'Następne >',
+      type: 'Typ'
+    },
+    dataset_grouping: {
+      'year': 'lat',
+      'month': 'miesięcy',
+      'week': 'tygodni',
+      'day': 'dni',
+      'hour': 'godzin'
+    },
+    dataset_type: {
+      'occupancy': 'zajętość',
+      'relative_occupancy': 'względna zajętość',
+      'average_dish_review': 'średnia ocena dania'
+    }
+  }
+}
+
+/* Date utilities */
+
+function monthLeadZero(date) {
+  return ('0' + (1 + date.getMonth())).slice(-2);
+}
+
+function dayLeadZero(date) {
+  return ('0' + date.getDate()).slice(-2);
+}
+
+function hourLeadZero(date) {
+  return ('0' + date.getHours()).slice(-2);
+}
+
+function minuteLeadZero(date) {
+  return ('0' + date.getMinutes()).slice(-2);
+}
+
+function dateYM(date) {
+  return date.getFullYear() + '-' + monthLeadZero(date);
+}
+
+function dateYMD(date) {
+  return date.getFullYear() + '-' + monthLeadZero(date) + '-' + dayLeadZero(date);
+}
+
+function dateYMDHM(date) {
+  return date.getFullYear() + '-' + monthLeadZero(date) + '-' + dayLeadZero(date) + ' ' + hourLeadZero(date) + ':' + minuteLeadZero(date);
+}
+
+/* Color utilities */
+
+class Color {
+  constructor(hex, r, g, b, h, s, l) {
+    this.hex = hex;
+    this.r = r;
+    this.g = g;
+    this.b = b;
+    this.h = h;
+    this.s = s;
+    this.l = l;
+  }
+
+  rgba(alpha) {
+    return 'rgba(' + this.r + ', ' + this.g + ', ' + this.b + ', ' + alpha + ')';
+  }
```

Comment:
what standard of JS do you use. If some newer let's use [templates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)